src/utils/permissions.ts

```diff
@@ -0,0 +1,321 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+
+import { ColonyRole, Id } from '@colony/colony-js';
+
+import { mutate, query } from '~amplifyClient';
+import { ContractEvent, ContractEventsSignatures, ColonyActionType } from '~types';
+
+import {
+  verbose,
+  getCachedColonyClient,
+  getDomainDatabaseId,
+  mapLogToContractEvent,
+  writeActionFromEvent,
+} from '~utils';
+
+const BASE_ROLES_MAP = {
+  [`role_${ColonyRole.Recovery}`]: null,
+  [`role_${ColonyRole.Root}`]: null,
+  [`role_${ColonyRole.Arbitration}`]: null,
+  [`role_${ColonyRole.Architecture}`]: null,
+  [`role_${ColonyRole.Funding}`]: null,
+  [`role_${ColonyRole.Administration}`]: null,
+};
+
+export const getColonyRolesDatabaseId = (
+  colonyAddress: string,
+  nativeDomainId: number,
+  userAddress: string,
+): string => `${colonyAddress}_${nativeDomainId}_${userAddress}_roles`;
+
+export const getColonyHistoricRolesDatabaseId = (
+  colonyAddress: string,
+  nativeDomainId: number,
+  userAddress: string,
+  blockNumber: number,
+): string => `${colonyAddress}_${nativeDomainId}_${userAddress}_${blockNumber}_roles`;
+
+export const getAllRoleEventsFromTransaction = async (
+  transactionHash: string,
+  colonyAddress: string,
+): Promise<ContractEvent[]> => {
+  const colonyRoleSetEventName = ContractEventsSignatures.ColonyRoleSet.slice(0, ContractEventsSignatures.ColonyRoleSet.indexOf('('));
+  const colonyRecoveryRoleSetName = ContractEventsSignatures.RecoveryRoleSet.slice(0, ContractEventsSignatures.RecoveryRoleSet.indexOf('('));
+
+  const colonyClient = await getCachedColonyClient(colonyAddress);
+  const transactionReceipt = await colonyClient.provider.getTransactionReceipt(transactionHash);
+
+  const events = await Promise.all(
+    transactionReceipt.logs.map(log => mapLogToContractEvent(log, colonyClient.interface)),
+  );
+
+  const filteredEvents = events.filter(event => {
```

Comment:
Any chance you could run prettier on this pr so we keep style consistent? I know there's an issue for that to be handled automatically in this repo, but for the meantime?

Reply:
This filter fn could be more succinct:

``` js
events.filter((event) => event?.name === colonyRoleSetEventName || event?.name === colonyRecoveryRoleSetName)
```